src/manifest.ts

```diff
@@ -462,14 +462,16 @@ export class Manifest {
      * @param  {string[]}   buildArgs   The build arguments
      * @param  {string}     configOpts  The configuration options
      */
-    getMesonCommands(
+    async getMesonCommands(
         rebuild: boolean,
         buildArgs: string[],
         configOpts: string[]
-    ): Command[] {
+    ): Promise<Command[]> {
+        const mesonBuildParentDir = (await findFileParent(this.workspace, 'meson.build'))?.toString() || this.workspace
+
```

Comment:
Can't we cache this value as the workspace the moment we construct the Manifest?

Reply:
We can't run async functions from the constructor, unless `findFileParent` is converted to a sync one.

Reply:
I think we should still check for .git as well. Because if you have the flatpak manifest in the data dir, the directory would contain a meson.build but not the correct one.

Reply:
The `walk` function also walks `.git` and other dotfiles.

Reply:
What I meant is we would retrieve the value here but cache it in a variable to avoid calling the findFileParent for every meson command we run.

Reply:
Also not sure if this would be specific to meson or to other build systems as well. I feel like we should base the parent dir on the resolved path of `dir` inside the manifest of the application module if it is set first and then fallback to our function, maybe?

Reply:
> Also not sure if this would be specific to meson or to other build systems as well. I feel like we should base the parent dir on the resolved path of dir inside the manifest of the application module if it is set first and then fallback to our function, maybe?

I do prefer that solution indeed, though I'm not sure how would we get the dir reliably

Would this be sufficient?

```ts
lastModuleParentDir(): PathLike | undefined {
    const lastModule = this.lastModule()

    if (lastModule.sources.length !== 1) {
        return undefined
    }

    const moduleLastSource = lastModule.sources.slice(-1)[0]

    if (moduleLastSource.type !== 'dir') {
        return undefined
    }

    if (moduleLastSource.path === undefined) {
        return undefined
    }

    return path.join(this.workspace, moduleLastSource.path.toString())
}
```

Reply:
Also, I think checking for `dir` on the manifest instead of finding the top level `meson.build` is less hacky since I guess it is always expected for a meson project to have `meson.build` on the project root as specified in the manifest, so we could just drop `findFileParent`